python/interpret_text/experimental/introspective_rationale/introspective_rationale_explainer.py

```diff
@@ -224,11 +224,19 @@ def explain_local(self, text: str, **kwargs) -> _create_local_explanation:
             seq_len = int(m.sum().item())
             ids = x[:seq_len][0]
         tokens = kwargs['preprocessor'].decode_single(ids)
-
+        local_importance_values = zs.flatten()
+        # post-processing for BERT to remove SEP and CLS tokens
+        # TODO: might we want to add a "post-process" method to the preprocessor?
+        tokens_to_remove = [BertTokens.SEP, BertTokens.CLS]
+        token_indexes = [idx for idx, token in enumerate(tokens) if token in tokens_to_remove]
+        if token_indexes:
+            local_importance_values = np.delete(local_importance_values, token_indexes)
```

Comment:
would:

local_importance_values = local_importance_values[token_indexes]  # Work?

Reply:
not quite, those are the indexes of tokens to delete, not the chosen indexes to keep

Reply:
If we flipped the condition?

Reply:
if token not in tokens_to_remove

Reply:
by flip you mean to generate all of the indexes to keep instead of the indexes to remove and then select those?  I think that's what numpy.delete does but in a more efficient manner:
https://github.com/numpy/numpy/blob/v1.17.0/numpy/lib/function_base.py#L4424

Reply:
I could do it the other way too, it's definitely not a performance bottleneck, but then tokens is a list and it doesn't have the sort of slicing by list of indexes that numpy has, although I could convert tokens from list to numpy and then back to list after doing the slicing by array of indexes but that seems a bit more messy looking stylistically

Reply:
Ok, if this works it's fine. I tend to avoid delete and follow up delete but if there isn't a better way we can leave it as such